### PR TITLE
adding design template

### PR DESCRIPTION
**Pull request by SchSeba**

### CODE REVIEW

**Review by zeeke — APPROVED** [review]
LGTM

File: doc/design/TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+## Proposal
+
+### Workflow Description
+
+### API Extensions
+
+### Implementation Details/Notes/Constraints
+
+### Test Plan
```
Comment (adrianchiris) [inline doc/design/TEMPLATE.md:31]: lets have : Update  & Backward Compatibility considerations

Comment (SchSeba) [reply doc/design/TEMPLATE.md:33]: I like it :)

**Comment by SchSeba** [comment]
done

File: doc/design/TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+## Proposal
+
+### Workflow Description
+
+### API Extensions
+
+### Implementation Details/Notes/Constraints
+
+### Update & Backward Compatibility considerations
```
Comment (adrianchiris) [inline doc/design/TEMPLATE.md:31]: nit: should it be Upgrade ? now that i think about it ?

Comment (adrianchiris) [reply doc/design/TEMPLATE.md:31]: ah i wrote upgrade in the original comment :) 

so which is better ?

**Pull request merged**